src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -163,60 +163,31 @@ static inline bool is_class_loader(const Symbol* class_name,
 // private: called to verify that k is a static member of this nest.
 // We know that k is an instance class in the same package and hence the
 // same classloader.
-bool InstanceKlass::has_nest_member(InstanceKlass* k, TRAPS) const {
+bool InstanceKlass::has_nest_member(InstanceKlass* k, JavaThread* current) const {
```

Comment:
Should we add an assert that "k is an instance class in the same package and hence the same classloader"

Reply:
I don't think we need to given there is only a single caller of this method and it happens after those very checks:

if (is_same_class_package(k)) {
        // Now check actual membership. We can't be a member if our "host" is
        // not an instance class.
        if (k->is_instance_klass()) {
          nest_host_k = InstanceKlass::cast(k);
          bool is_member = nest_host_k->has_nest_member(this, current);